File: ThreadLab4C#/ThreadLab4/ThreadLab4WithMonitor/Philosopher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ThreadLab4
{
    public class Philosopher
    {
        private readonly Table _table;
        private readonly int _id;

        public Philosopher(Table table, int id)
        {
            _table = table;
            _id = id;
        }

        public void Run()
        {
            for (int i = 0; i < 10; i++)
            {
                Console.WriteLine("Philosopher " + _id + " is thinking " + (i + 1) + " times");
                _table.TakeForks(_id);
                Console.WriteLine("Philosopher " + _id + " is eating " + (i + 1) + " times");
                _table.PutForks(_id);
            }
        }

    }
}

File: ThreadLab4Java/ThreadLab4WithMonitor/src/Philosopher.java
public class Philosopher implements Runnable {
    private final Table table;
    private final int id;

    public Philosopher(Table table, int id) {
        this.table = table;
        this.id = id;
    }

    @Override
    public void run() {
        for (int i = 0; i < 10; i++) {
            System.out.println("Philosopher " + id + " is thinking " + (i + 1) + " times");
            table.takeForks(id);
            System.out.println("Philosopher " + id + " is eating " + (i + 1) + " times");
            table.putForks(id);
        }
        
    }
}